Source/Dafny/Resolver.cs

```diff
@@ -11942,6 +11959,47 @@ private class RBranchExpr : RBranch {
           "Internal Error: Unknown failure type in :- statement");
         return;
       }
+
+      Expression lhsExtract = null;
+      if (expectExtract) {
+        Method caller = codeContext as Method;
+        if (caller != null && caller.Outs.Count == 0 && s.ExpectToken == null) {
+          reporter.Error(MessageSource.Resolver, s.Rhs.tok, "Expected {0} to have a Success/Failure output value",
+            caller.Name);
+          return;
+        }
+
+        lhsExtract = s.Lhss[0];
+        var lhsResolved = s.Lhss[0].Resolved;
+        // Make a new unresolved expression
+        if (lhsResolved is MemberSelectExpr lexr) {
+          Expression id = makeTemp("recv", s, codeContext, lexr.Obj);
+          var lex = lhsExtract as ExprDotName; // might be just a NameSegment
+          lhsExtract = new ExprDotName(lexr.tok, id, lexr.MemberName, lex == null ? null : lex.OptTypeArguments);
+        } else if (lhsResolved is SeqSelectExpr lseq) {
+          Expression id = makeTemp("recv", s, codeContext, lseq.Seq);
+          Expression id0 = null;
+          if (lseq.E0 != null) {
+            id0 = makeTemp("idx", s, codeContext, lseq.E0);
+          }
+          Contract.Assert(lseq.E1 == null);
```

Comment:
In the same style, `lseq.E0` can be asserted to be non-null (since `a[..]` is not a legal l-value)

Reply:
Done